src/httomolib/prep/normalize.py

```diff
@@ -48,15 +51,23 @@ def normalize_raw_cuda(
         3D flat field data as a CuPy array.
     darks : cp.ndarray
         3D dark field data as a CuPy array.
+    gpu_id : int, optional
+        A GPU device index to perform operation on.        
     cutoff : float, optional
         Permitted maximum value for the normalised data.
+    minus_log : bool, optional
+        Apply negative log to the normalised data.
+    nonnegativity : bool, optional
+        Remove negative values in the normalised data.          
 
     Returns
     -------
     cp.ndarray
         Normalised 3D tomographic data as a CuPy array.
     """
 
+    cp.cuda.Device(gpu_id).use()
+    
```

Comment:
```suggestion

    if data.ndim != 3:
        raise ValueError("Input data must be a 3D stack of projections")
```
Similar to `normalize_cupy`.